finesse/gui/temp_control.py

```diff
@@ -0,0 +1,311 @@
+
+        self._canvas.draw()
+
+    def _toggle_axis_visibility(self, name: str) -> None:
+        """Shows or hides BB plots."""
+        state = self._ax[name].yaxis.get_visible()
+        self._btns[name].setFlat(state)
+        self._ax[name].yaxis.set_visible(not state)
+        self._ax[name].lines[0].set_visible(not state)
+        self._canvas.draw()
+
+    def _update_figure(self, event) -> None:
+        """Updates the matplotlib figure to be contained within the panel."""
+        time = list(self._ax["hot"].lines[0].get_xdata())
+        hot_data = list(self._ax["hot"].lines[0].get_ydata())
+        cold_data = list(self._ax["cold"].lines[0].get_ydata())
+
+        time.pop(0)
+        hot_data.pop(0)
+        cold_data.pop(0)
+
+        new_time, new_hot_data, new_cold_data = get_temperature_data()
+        time.append(new_time)
+        hot_data.append(new_hot_data)
+        cold_data.append(new_cold_data)
+
+        self._ax["hot"].lines[0].set_xdata(time)
+        self._ax["hot"].lines[0].set_ydata(hot_data)
+        self._ax["cold"].lines[0].set_xdata(time)
+        self._ax["cold"].lines[0].set_ydata(cold_data)
+
+        self._ax["hot"].relim()
+        self._ax["cold"].relim()
+        self._ax["hot"].autoscale()
+        self._ax["cold"].autoscale()
+        self._ax["cold"].set_ylim(  # Confines "cold" line to lower half of plot
+            [self._ax["cold"].get_ylim()[0], 2 * self._ax["cold"].get_ylim()[1]]
+        )
+
+        xticks = self._ax["hot"].get_xticks()
+        xticklabels = [""] * len(xticks)
+        for i in range(len(xticks)):
+            t = datetime.fromtimestamp(xticks[i])
+            xticklabels[i] = t.strftime("%H:%M:%S")
+        #        self._ax["cold"].set_xticks(xticks)
+        self._ax["hot"].set_xticklabels(xticklabels)
+
+        self._canvas.draw()
+
+
+class DP9800(QGroupBox):
+    """Widgets to view the DP9800 properties."""
+
+    def __init__(self, num_channels: int) -> None:
+        """Creates the widgets to monitor DP9800.
+
+        Args:
+            num_channels (int): Number of Pt 100 channels being monitored
+        """
+        super().__init__("DP9800")
+
+        self._num_channels = num_channels
+
+        layout = self._create_controls()
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+        """Creates the overall layout for the panel.
+
+        Returns:
+            QGridLayout: The layout containing the figure.
+        """
+        layout = QGridLayout()
+
+        layout.addWidget(QLabel("Pt 100"), 1, 0)
+        self._channels = []
+        for i in range(self._num_channels):
+            label = QLabel("CH_%d" % (i + 1))
+            label.setAlignment(Qt.AlignmentFlag.AlignCenter)
+            layout.addWidget(label, 0, i + 1)
+            tbox = QLineEdit()
+            self._channels.append(tbox)
+            self._channels[i].setReadOnly(True)
+            layout.addWidget(self._channels[i], 1, i + 1)
```

Comment:
A couple of things:

- You can use an `f`-string for the `QLabel`
- You refer to the same thing as `tbox` and `self._channels[i]`

```suggestion
        self._channels = []
        for i in range(self._num_channels):
            label = QLabel(f"CH_{i+1}")
            label.setAlignment(Qt.AlignmentFlag.AlignCenter)
            layout.addWidget(label, 0, i + 1)

            channel = QLineEdit()
            channel.setReadOnly(True)
            layout.addWidget(channel, 1, i + 1)
            self._channels.append(channel)
```